### PR TITLE
Compare UUIDs instead of Strings.

### LINKED ISSUE #829: Compare UUIDs instead of Strings in `EntitySource.findFirstEntityByUuid`
Once #773 is merged:

This could be a quick one. I think it's more efficient to compare UUIDs in `findFirstEntityByUuid` instead of Strings (UUIDs just consist of two longs).

_Originally mentioned by @sebastian-peter in https://github.com/ie3-institute/PowerSystemDataModel/pull/773#discussion_r1211379308_

### PR DESCRIPTION
**Pull request by staudtMarius**
Resolves #829

### CODE REVIEW

**Comment by staudtMarius** [comment]
!test

File: src/main/java/edu/ie3/datamodel/io/source/EntitySource.java [hunk 1]
```diff
@@ -71,10 +71,10 @@ protected String safeMapGet(Map<String, String> map, String key, String mapName)
    *     optional if no matching entity with the provided uuid can be found
    */
   protected <T extends UniqueEntity> Optional<T> findFirstEntityByUuid(
-      String entityUuid, Collection<T> entities) {
+      UUID entityUuid, Collection<T> entities) {
     return entities.stream()
         .parallel()
-        .filter(uniqueEntity -> uniqueEntity.getUuid().toString().equalsIgnoreCase(entityUuid))
+        .filter(uniqueEntity -> uniqueEntity.getUuid().equals(entityUuid))
```
Comment (sebastian-peter) [inline src/main/java/edu/ie3/datamodel/io/source/EntitySource.java:77]: Okay what one should actually do (at least for frequently searched entity sets like nodes) is build a HashMap uuid -> entity, which is way faster. But that's way beyond the scope of this PR.

**Review by sebastian-peter — CHANGES_REQUESTED** [review]
Two tiny remarks from my side.

File: src/main/java/edu/ie3/datamodel/io/source/EntitySource.java (hunk 1 above)
Comment (staudtMarius) [reply src/main/java/edu/ie3/datamodel/io/source/EntitySource.java:77]: Should I change something in this line or are you creating a new issue for this?

Comment (sebastian-peter) [reply src/main/java/edu/ie3/datamodel/io/source/EntitySource.java:77]: Honestly I'd just leave it as it is for now, and I don't think we need an issue for this yet. If we want to optimize for speed at some point, we should measure performance (with profiling or such) anyways.

**Review by sebastian-peter — APPROVED** [review]
Thanks, looks good!

**Pull request merged**